crypto/fipsmodule/cipher/cipher.c

```diff
@@ -590,6 +588,20 @@ unsigned EVP_CIPHER_CTX_key_length(const EVP_CIPHER_CTX *ctx) {
   return ctx->key_len;
 }
 
+int EVP_CIPHER_CTX_get_iv(const EVP_CIPHER_CTX *ctx, unsigned char *iv,
+                          size_t len) {
+  if (ctx == NULL || iv == NULL)
+    return 0;
+
+  if (len > EVP_MAX_IV_LENGTH || len != EVP_CIPHER_CTX_iv_length(ctx))
```

Comment:
It shouldn't be possible. I agree this might be a slightly excessive use of "defensive programming".  

The concern is that a client allocates a static buffer of size `EVP_MAX_IV_LENGTH` to use for the IV under the (hopefully correct) assumption that it can never be bigger -- the client might not even bother to verify that this "max" size is honored. If there is a bug elsewhere in our logic that might allow the IV length to be larger than this, we could overflow the client's allocation when we copy the value.